File: src/main/java/commands/slashadapters/adapters/ClearAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.moderationcategory.ClearCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

@Slash(command = ClearCommand.class)
public class ClearAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.INTEGER, "amount", "How many messages shall be removed? (2 - 500)", true);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(ClearCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/ClearAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.moderationcategory.ClearCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

@Slash(command = ClearCommand::class)
class ClearAdapter : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData.addOptions(
            generateOptionData(OptionType.INTEGER, "amount", "clear_amount", true),
            generateOptionData(OptionType.CHANNEL, "channel", "clear_channel", false),
            generateOptionData(OptionType.STRING, "members", "clear_members", false)
        )
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(ClearCommand::class.java, collectArgs(event))
    }

}